### ISSUE TITLE
1.x: Flaky test OperatorObserveOnTest.testQueueFullEmitsErrorWithVaryingBufferSize

### ISSUE THREAD

**Issue by artem-zinnatullin** [ReactiveX/RxJava#3819]
Fails constantly on my dev machine today and now I [saw it on Travis](https://travis-ci.org/ReactiveX/RxJava/builds/119904219).

```
rx.internal.operators.OperatorObserveOnTest > testQueueFullEmitsErrorWithVaryingBufferSize FAILED
    java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:93)
        at org.junit.Assert.failNotEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:128)
        at org.junit.Assert.assertEquals(Assert.java:472)
        at org.junit.Assert.assertEquals(Assert.java:456)
        at rx.internal.operators.OperatorObserveOnTest.testQueueFullEmitsErrorWithVaryingBufferSize(OperatorObserveOnTest.java:631)
```

**Comment by akarnokd** [comment]
Yep, fixed in #3795 awaiting review...

**Comment by akarnokd** [comment]
Fixed separately in #3825.
